common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
I should tell you that we generate the tracing information (that I think is covered by this pr) already by post analysis of the logs. @jeremiaswerner maybe you want to post/share a sample?

We could change the Message to have an optional metadata field and provide a heck to have it decorated by the tracing framework when one is available. Although _this_ will still be on the critical path where we are trying to focus and get leaner. So we need to be cognizant of that.